client/mm/exchange_adaptor.go

```diff
@@ -2428,14 +2429,17 @@ func (u *unifiedExchangeAdaptor) tryCancelOrders(ctx context.Context, epoch *uin
 			continue
 		}
 
-		done = false
 		err = u.CEX.CancelTrade(ctx, baseID, quoteID, tradeID)
 		if err != nil {
 			u.log.Errorf("Error canceling CEX trade %s: %v", tradeID, err)
+		} else {
+			cexCancels++
 		}
 	}
 
-	return done
+	totalCancels := dexCancels + cexCancels
+	expectedCancels := len(u.pendingDEXOrders) + len(u.pendingCEXOrders)
+	return totalCancels == expectedCancels
```

Comment:
This looks different. It was checking order status, not the number of all orders. This will have more expectedCancels than is really expected. They aren't all expected to be canceled.

Reply:
i.e expectedCancels = number of uncompleted  trades(if cex) + number of orders in the free cancel orders (dex)

We need to be able to keep track of them in this function to be able to accurately report to the caller whether or not all “cancel-able”orders were actually cancelled.

Will update this.

Reply:
I'm don't understand what the problem is. `cancels` looks unused, so remove that, but I don't see why the rest was wrong.

Reply:
maybe @martonp can comment if they see this

Reply:
The function `tryCancelOrders` cancels all cancellable orders, but it reports when no further orders remain to be canceled. Its return value is used in `cancelAllOrders` to signal that the function can complete and the bot can shut down. If there are still orders on the books that were placed too recently to be cancelled, we don't want to shut down the bot yet.

Reply:
Why are you changing this btw? Did you encounter any issues related to this?

Reply:
@martonp said this can be removed, unsure though.

```
if !cancelCEXOrders {
		return false
}
```

Should it return true if all the dex orders are canceled then instead of false?

I guess I see the initial problem, you want to still return true if we are able to cancel the cancel-able ones?

Also, I don't think there needs to be two `successfulXXX` variables, cant they be combined like expected?

Reply:
From the way the method is used, we can’t remove that early return and I think the rationale for returning false is because there might be some cancel-able CEX orders.

Yes.

Errm, yes.

Reply:
Sorry, my mistake, that block should not be removed.
The current update is not correct though, because don't to be returning true if all cancellable orders were cancelled. We want to be returning true when all the orders have already been cancelled.

Reply:
Ok, sorry for the long back and forth, but I think all that needs to change is the removal of the unused slice, the done variable was fine.

Reply:
I will revert the changes in this function, I think @martonp can look through it at a later time to see what needs improvement. 

IMO, done is never true unless cancel was not tried on any order.

Reply:
it starts true

Reply:
> IMO, done is never true unless cancel was not tried on any order.

Yes, that's the point.

Reply:
Ah okay…gotcha.